packages/app-layout/AppLayout.tsx

```diff
@@ -0,0 +1,161 @@
+import { h, Component, prop } from 'skatejs';
+
+import {
+  GenericTypes,
+  matchMedia
+} from '../_helpers';
+
+import { HamburgerButton } from './components/HamburgerButton';
+import { Drawer } from './components/Drawer';
+import { Overlay } from './components/Overlay';
+import { Nav } from './components/Nav';
+import { NavContent } from './components/NavContent';
+
+import styles from './AppLayout.scss';
+
+const DEFAULT_RESPONSIBLE_WIDTH = '640px';
+
+type AppLayoutProps = Props & EventProps;
+
+type Attrs = {
+  'drawer-visible'?: boolean,
+  'responsive-width'?: string
+};
+
+type Props = {
+  drawerVisible?: boolean,
+  responsiveWidth?: string
+};
+
+// TODO: add onDrawerChange / onNarrowChange
+type EventProps = {};
+
+type Events = {};
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-app-layout': GenericTypes.IntrinsicCustomElement<AppLayoutProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}
+
+export class AppLayout extends Component<AppLayoutProps> {
+
+  static get is() { return 'bl-app-layout'; }
+
+  static get props() {
+    return {
+      drawerVisible: prop.boolean( {
+        attribute: {
+          source: true
+        }
+      } ),
+      responsiveWidth: prop.string( {
+        initial: DEFAULT_RESPONSIBLE_WIDTH,
+        attribute: {
+          source: true
+        }
+      } ),
+      narrow: prop.boolean( {
+        attribute: {
+          source: true,
```

Comment:
I like code to be more explicit(like separating to getSetProps) as arity is not that obvious from first time reader point of view also seems even tray use it that way https://github.com/skatejs/skatejs/blob/7db2988c4d6c20651e8dad2145912ee3bea659e4/src/api/component.js#L22 :). 

Anyway that's just nitpicking, I'm bit confused from asynchronicity of props. As you are very well aware of setAsyncState in react does this count for skate as well? What I have got from your note, it's async when done directly, but using props setter is sync. Does it mean that we can count on that and use this.someProp as getter (as we are suppose to use props setter every time) or we should rather use pops getter as well? Are there any cases where we should rather use async setter?